ui/src/app/shared/components/tags-input/tags-input.tsx

```diff
@@ -0,0 +1,113 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+
+import {Autocomplete, AutocompleteApi, AutocompleteOption} from 'argo-ui';
+
+export interface TagsInputProps {
+    tags: string[];
+    autocomplete?: (AutocompleteOption | string)[];
+    onChange?: (tags: string[]) => void;
+    placeholder?: string;
+}
+
+require('./tags-input.scss');
+
+export class TagsInput extends React.Component<TagsInputProps, {tags: string[]; input: string; focused: boolean}> {
+    private inputEl: HTMLInputElement;
+    private autocompleteApi: AutocompleteApi;
+
+    constructor(props: TagsInputProps) {
+        super(props);
+        this.state = {tags: props.tags || [], input: '', focused: false};
+    }
+
+    public render() {
+        return (
+            <div
+                className={classNames('tags-input argo-field', {'tags-input--focused': this.state.focused || !!this.state.input})}
+                onClick={() => this.inputEl && this.inputEl.focus()}>
+                {this.state.tags.map((tag, i) => (
+                    <span className='tags-input__tag' key={tag}>
+                        {tag}{' '}
+                        <i
+                            className='fa fa-times'
+                            onClick={e => {
+                                const newTags = [...this.state.tags.slice(0, i), ...this.state.tags.slice(i + 1)];
+                                this.setState({tags: newTags});
+                                this.onTagsChange(newTags);
+                                e.stopPropagation();
+                            }}
+                        />
+                    </span>
+                ))}
+                <Autocomplete
+                    filterSuggestions={true}
+                    autoCompleteRef={api => (this.autocompleteApi = api)}
+                    value={this.state.input}
+                    items={this.props.autocomplete}
+                    onChange={e => this.setState({input: e.target.value})}
+                    onSelect={value => {
+                        if (this.state.tags.indexOf(value) === -1) {
+                            const newTags = this.state.tags.concat(value);
+                            this.setState({input: '', tags: newTags});
+                            this.onTagsChange(newTags);
+                        }
+                    }}
+                    renderInput={props => (
+                        <input
```

Comment:
This whole `<input>` field doesn't seem like it is the correct way to do this. Can you check how we take input elsewhere in the codebase?